Index: share/jive/applets/SlimMenus/SlimMenusApplet.lua
===================================================================
--- share/jive/applets/SlimMenus/SlimMenusApplet.lua	(revision 783)
+++ share/jive/applets/SlimMenus/SlimMenusApplet.lua	(working copy)
@@ -621,7 +621,7 @@
 			elseif item.id == "music_services" or item.id == "_music_services" then
 				--ignore, handled with app store
 
-			elseif item.id == "settingsAlarm" then
+			-- elseif item.id == "settingsAlarm" then
 				--ignore as we can't render time input at present
 				-- FIXME - remove when time input can be rendered in HD skins
 
