Index: share/jive/jive/JiveMain.lua
===================================================================
--- share/jive/jive/JiveMain.lua	(revision 783)
+++ share/jive/jive/JiveMain.lua	(working copy)
@@ -18,6 +18,9 @@
 =cut
 --]]
 
+-----------------------------------------------------------------------------------------
+local io            = require("io")
+-----------------------------------------------------------------------------------------
 
 -- stuff we use
 local math          = require("math")
@@ -195,7 +198,19 @@
 		--already in the desired state, leave (can happen for instance when notify_playerPower comes back after a local power change)
 		 return
 	end
+-----------------------------------------------------------------------------------------
+	local isPiDisplay = _read("/sys/class/backlight/rpi_backlight/brightness")
+	
+	local backlightBrightness = appletManager:callService("getBacklightBrightnessWhenOn")
+	if backlightBrightness == nil then
+		backlightBrightness = "255"
+	end
 
+	local reducedBacklightBrightness = appletManager:callService("getBacklightBrightnessWhenOff")
+	if reducedBacklightBrightness == nil then
+		reducedBacklightBrightness = "130"
+	end
+-----------------------------------------------------------------------------------------
 	_softPowerState = softPowerState
 	local currentPlayer = appletManager:callService("getCurrentPlayer")
 	if _softPowerState == "off" then
@@ -203,6 +218,11 @@
 		if currentPlayer and (currentPlayer:isConnected() or currentPlayer:isLocal()) then
 			currentPlayer:setPower(false, nil, isServerRequest)
 		end
+-----------------------------------------------------------------------------------------
+		if isPiDisplay ~= nil then 
+			_write("/sys/class/backlight/rpi_backlight/brightness", reducedBacklightBrightness)
+		end
+-----------------------------------------------------------------------------------------
 		--todo: also pause/power off local player since local player might be playing and not be the current player
 		appletManager:callService("activateScreensaver", isServerRequest)
 	elseif _softPowerState == "on" then
@@ -215,7 +235,11 @@
 			end
 			currentPlayer:setPower(true, nil, isServerRequest)
 		end
-
+-----------------------------------------------------------------------------------------
+		if isPiDisplay ~= nil then 
+			_write("/sys/class/backlight/rpi_backlight/brightness", backlightBrightness)
+		end
+-----------------------------------------------------------------------------------------
 		appletManager:callService("deactivateScreensaver")
 		appletManager:callService("restartScreenSaverTimer")
 
@@ -224,6 +248,27 @@
 	end
 end
 
+-----------------------------------------------------------------------------------------
+function _write(file, val)
+	local fh, err = io.open(file, "w")
+	if err then
+		return
+	end
+	fh:write(val)
+	fh:close()
+end
+
+function _read(file)
+	local fh, err = io.open(file, "r")
+	if err then
+		return nil
+	end
+	local fc = fh:read("*all")
+	fh:close()
+	return fc
+end
+-----------------------------------------------------------------------------------------
+
 function JiveMain:togglePower()
 	local powerState = JiveMain:getSoftPowerState()
 	if powerState == "off" then
