--- share/jive/jive/System.lua.orig	2024-02-03 16:10:50.377804656 -0500
+++ share/jive/jive/System.lua	2024-02-03 16:21:08.532078038 -0500
@@ -38,7 +38,9 @@
 
 local oo           = require("loop.simple")
 local log           = require("jive.utils.log").logger("jivelite")
-
+--start pCP patch----------------------------------------------------
+local rpi          = require("jive.utils.rpi_bl") 
+--end pCP patch------------------------------------------------------
 
 -- our class
 module(...)
@@ -96,9 +98,15 @@
 	return _capabilities["hasDigitalOut"] ~= nil
 end
 
+--start pCP patch----------------------------------------------------
 function hasTouch(self)
-	return _capabilities["touch"] ~= nil
+    if rpi.isTouch() ~= nil then
+        return true
+    else
+        return _capabilities["touch"] ~= nil
+    end
 end
+--end pCP patch------------------------------------------------------
 
 function hasIr(self)
 	return _capabilities["ir"] ~= nil
